CoreScriptsRoot/Modules/Server/ClientChat/ChatBar.lua

```diff
@@ -113,9 +121,31 @@ function methods:CreateGuiObjects(targetParent)
 
 	self:AnimGuiObjects()
 	self:SetUpTextBoxEvents(TextBox, TextLabel, MessageModeTextButton)
+	if self.UserHasChatOff then
+		self:DoLockChatBar()
+	end
 	self.eGuiObjectsChanged:Fire()
 end
 
+-- Used to lock the chat bar when the user has chat turned off.
+function methods:DoLockChatBar()
+	if self.TextLabel then
+		if LocalPlayer.UserId > 0 then
+			self.TextLabel.Text = "To chat in game, turn on chat in your Privacy Settings."
+		else
+			self.TextLabel.Text = "Sign up to chat in game."
+		end
+		self:CalculateSize()
+	end
+	if self.TextBox then
+		self.TextBox.Active = false
+		local textboxfocusedConnection = self.TextBox.Focused:connect(function()
+			self.TextBox:ReleaseFocus()
+		end)
+		table.insert(self.Connections, textboxfocusedConnection)
```

Comment:
whats the point in storing this connection? do all the connections get destroyed when something in here is destroyed?